mypy/stubinfo.py

```diff
@@ -1,22 +1,53 @@
 from __future__ import annotations
 
 
-def is_legacy_bundled_package(prefix: str) -> bool:
-    return prefix in legacy_bundled_packages
+def is_module_from_legacy_bundled_package(module: str) -> bool:
+    top_level = module.split(".")[0]
+    return top_level in legacy_bundled_packages
 
 
-def approved_stub_package_exists(prefix: str) -> bool:
-    return is_legacy_bundled_package(prefix) or prefix in non_bundled_packages
+def approved_stub_package_exists(module: str) -> bool:
+    components = module.split(".")
```

Comment:
Again, it might be slightly faster to use `module.split(".", 1)[0]` to calculate prefix, and only do the full split in the body of the last if statement as needed.